freeciv-web/src/main/webapp/javascript/control.js

```diff
@@ -7675,6 +7675,22 @@ function openFullscreen() {
     }
   });
 }
+
+/****************************************************************************
+ This progressive enhancement changes full screen mode to require a long
+ press of the [Esc] key, so that the player doesn't accidentally leave it.
```

Comment:
```suggestion
 press of the [Esc] key to exit, so that the player doesn't accidentally leave full
 screen mode, but can instead cancel in-game dialogs by pressing [Esc].
```